packages/playwright-core/src/server/utils/network.ts

```diff
@@ -137,23 +132,27 @@ export function createProxyAgent(proxy?: ProxySettings, forUrl?: URL) {
   if (!/^\w+:\/\//.test(proxyServer))
     proxyServer = 'http://' + proxyServer;
 
-  const proxyOpts = url.parse(proxyServer);
+  const proxyOpts = new URL(proxyServer);
   if (proxyOpts.protocol?.startsWith('socks')) {
     return new SocksProxyAgent({
       host: proxyOpts.hostname,
       port: proxyOpts.port || undefined,
     });
   }
-  if (proxy.username)
-    proxyOpts.auth = `${proxy.username}:${proxy.password || ''}`;
+  if (proxy.username) {
+    proxyOpts.username = proxy.username;
+    proxyOpts.password = proxy.password || '';
+  }
 
   if (forUrl && ['ws:', 'wss:'].includes(forUrl.protocol)) {
     // Force CONNECT method for WebSockets.
-    return new HttpsProxyAgent(proxyOpts);
+    // TODO: switch to URL instance instead of legacy object once https-proxy-agent supports it.
+    return new HttpsProxyAgent(convertURLtoLegacyUrl(proxyOpts));
   }
 
   // TODO: We should use HttpProxyAgent conditional on proxyOpts.protocol instead of always using CONNECT method.
-  return new HttpsProxyAgent(proxyOpts);
+  // TODO: switch to URL instance instead of legacy object once https-proxy-agent supports it.
+  return new HttpsProxyAgent(convertURLtoLegacyUrl(proxyOpts));
```

Comment:
We are still on and old https-proxy-agent version which uses `url.parse` internally if you pass a string - once we update we can remove the `convertURLtoLegacyUrl` function and pass the URL object over.